### ISSUE TITLE
Using dynamic configuration map instead of configuration file not working

### ISSUE THREAD

**Issue by kunalransing** [paypal/PayPal-Java-SDK#73]
Description:
When we use below code (<a href='https://devtools-paypal.com/guide/pay_paypal/java?interactive=ON&env=sandbox'> https://devtools-paypal.com/guide/pay_paypal/java?interactive=ON&env=sandbox</a>)
```
Map<String, String> sdkConfig = new HashMap<String, String>();
sdkConfig.put("mode", "sandbox");
String accessToken = new OAuthTokenCredential("xxxx", "xxxx", sdkConfig).getAccessToken();
```
Now when we use above accessToken for payment.create(accessToken )..gives exception 
```
java.lang.RuntimeException: java.io.IOException: Resource 'sdk_config.properties' could not be found.
```
I think, when we use dynamic configuration map...we should not need 'sdk_config.properties'.

<b>Reason:</b>
When we Payment.initConfig() or PayPalResource.initConfig(),it internally set private field of PaypalResource 'configInitialized'=true.
But if we don't  PayPalResource.initConfig() then that field's default value is false.
So when call payment.create(accessToken) internally it call PayPalResource.configureAndExecute() where below code in PayPalResource.configureAndExecute() try to load default config file
                                if (!configInitialized) 
                                {
					initializeToDefault();
				}

**Comment by juwlee** [comment]
@kunalransing thanks for the suggestion. It would be great if you submit this kind of suggestion in the form of pull request. Would you like to do so?

**Comment by kunalransing** [comment]
Ok. I will commit changed code.

**Comment by juwlee** [comment]
@kunalransing any update? Also, can you clarify how you are loading config, if not using initConfig()?

**Comment by richardrega** [comment]
I'm not using a config file, but

		Map<String, String> sdkConfig = new HashMap<String, String>();
		sdkConfig.put("mode", Mode);
		sdkConfig.put("http.ConnectionTimeOut", "5000");
		sdkConfig.put("http.Retry", "2");
		sdkConfig.put("http.ReadTimeOut", "30000");
		sdkConfig.put("http.MaxConnection", "100");
		sdkConfig.put("http.ProxyPort", "8080");
		sdkConfig.put("http.ProxyHost", "127.0.0.1");
		sdkConfig.put("http.UseProxy", "false");
		sdkConfig.put("http.ProxyUserName", "null");
		sdkConfig.put("http.ProxyPassword", "null");
		sdkConfig.put("http.GoogleAppEngine", "false");
		sdkConfig.put("service.EndPoint", "https://" + Endpoint);
		sdkConfig.put("clientID", ClientID);
		sdkConfig.put("clientSecret", Secret);
		Properties properties = new Properties();
		properties.putAll(sdkConfig);
		Payment.initConfig(properties);

was enough to get around that for now.

I've also found that functions like FuturePayment's getCredential() don't work because I'm using this config instead of an actual file...

**Comment by kunalransing** [comment]
I am really sorry for no reply since then.
I am using below way which work fine. And I don't think to do any changes.
	Map<String, String> sdkConfig = new HashMap<String, String>();
    	sdkConfig.put("mode",mode);
    	OAuthTokenCredential authTokenCredential = new        OAuthTokenCredential(clientId,secret,sdkConfig);	//I taken clientId, secret from DB.
    	APIContext apiContext = new APIContext(authTokenCredential.getAccessToken());
    	apiContext.setConfigurationMap(sdkConfig);

and for every call(Not checked FuturePayment) I used apicontext like <b>payment.create(apiContext);</b>

**Comment by jaypatel512** [comment]
Hey @kunalransing ! 

What if we introduced the getCredential object to have apiContext as one of the params, and we get the credentials from there, which would resolve the issue you are mentioning. 

However, this is something that may need some more time to test it out, and a bit of refactoring too. I will add it to the to-do list, and see if I can help you with that. In the mean time, please use the suggestion you have provided, and you should be fine. 

Thanks.

**Comment by kunalransing** [comment]
Thank you for your suggestion.

**Comment by jaypatel512** [comment]
Hey @kunalransing , @richardrega !

We are working really hard on making some really interesting changes to APIContext class to make integration a piece of 🍰 .

I have created a work in progress PR: https://github.com/paypal/PayPal-Java-SDK/pull/191 that includes changes to the APIContext, and how the newer version of integration would look like. This PR would:

- Remove the need for sdk_config.properties.
- Make APIContext initialization really simple.
- Make future payment/third party invoicing integration that includes refresh token extremely simple.
- Improve the code readability.

I request you to please comment on the PR, and let us know your thoughts on it, and any improvements that we may have missed.

**Comment by jaypatel512** [comment]
Hey @kunalransing @richardrega !

We have finally released version [1.8.0](https://github.com/paypal/PayPal-Java-SDK/releases/tag/v1.8.0), that includes the easiest and simplest way of integrating PayPal Java SDK. 

Please upload your dependencies, and let us know any feedback, bugs, or improvements you would like to see in this release.
